azure-mgmt-dns/src/main/java/com/microsoft/azure/management/dns/DnsRecordSet.java

```diff
@@ -138,6 +147,37 @@
         }
 
         /**
+         * The first stage of a CNAME record set definition.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface CNameRecordSetBlank<ParentT> extends WithCNameRecordAlias<ParentT> {
+        }
+
+        /**
+         * The stage of a CNAME record definition allowing to add alias.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithCNameRecordAlias<ParentT> {
+            /**
+             * Creates a CNAME record with the provided alias.
+             *
+             * @param alias the alias
+             * @return the next stage of the definition
+             */
+            WithCNameRecordSetAttachable<ParentT> withAlias(String alias);
```

Comment:
I think on any added methods or withers to existing released models, it'd be most discoverable to put the `Beta` on the most specific pieces as well, in this case this individual wither. I don't think users think users think of our fluent interfaces (`WithFoo`) as an interface or anything they should be paying attention to, so I'd just put the `Beta` on the new methods since that's what they're going to be interacting with and hovering their mouse over as they type.